itests/ddo_test.go

```diff
@@ -0,0 +1,170 @@
+package itests
+
+import (
+	"context"
+	"fmt"
+	"github.com/filecoin-project/boost/cmd/boost/util"
+	"github.com/filecoin-project/boost/itests/framework"
+	"github.com/filecoin-project/boost/storagemarket"
+	smtypes "github.com/filecoin-project/boost/storagemarket/types"
+	"github.com/filecoin-project/boost/testutil"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	verifregst "github.com/filecoin-project/go-state-types/builtin/v9/verifreg"
+	verifregtypes "github.com/filecoin-project/go-state-types/builtin/v9/verifreg"
+	lapi "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/verifreg"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/wallet/key"
+	"github.com/filecoin-project/lotus/itests/kit"
+	"github.com/google/uuid"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestDirectDeal(t *testing.T) {
+	ctx := context.Background()
+
+	kit.QuietMiningLogs()
+	framework.SetLogLevel()
+	_ = logging.SetLogLevel("fxlog", "WARN")
+
+	rootKey, err := key.GenerateKey(types.KTSecp256k1)
+	require.NoError(t, err)
+
+	verifierKey, err := key.GenerateKey(types.KTSecp256k1)
+	require.NoError(t, err)
+
+	bal, err := types.ParseFIL("100fil")
+	require.NoError(t, err)
+
+	var opts []framework.FrameworkOpts
+	opts = append(opts, framework.EnableLegacyDeals(true))
+	opts = append(opts, framework.EnsembleOpts(
+		kit.RootVerifier(rootKey, abi.NewTokenAmount(bal.Int64())),
+		// assign some balance to the verifier so they can send an AddClient message.
+		kit.Account(verifierKey, abi.NewTokenAmount(bal.Int64())),
+	))
+	f := framework.NewTestFramework(ctx, t, opts...)
+	err = f.Start()
+	require.NoError(t, err)
+	defer f.Stop()
+
+	vrh, err := f.FullNode.StateVerifiedRegistryRootKey(ctx, types.TipSetKey{})
+	require.NoError(t, err)
+	t.Log("Verified Registry Root Key", vrh.String())
+
+	// import the root key.
+	rootAddr, err := f.FullNode.WalletImport(ctx, &rootKey.KeyInfo)
+	require.NoError(t, err)
+
+	// import the verifier's key.
+	verifierAddr, err := f.FullNode.WalletImport(ctx, &verifierKey.KeyInfo)
+	require.NoError(t, err)
+
+	params, err := actors.SerializeParams(&verifregtypes.AddVerifierParams{Address: verifierAddr, Allowance: big.NewInt(100000000000)})
+	require.NoError(t, err)
+
+	msg := &types.Message{
+		From:   rootAddr,
+		To:     verifreg.Address,
+		Method: verifreg.Methods.AddVerifier,
+		Params: params,
+		Value:  big.Zero(),
+	}
+
+	sm, err := f.FullNode.MpoolPushMessage(ctx, msg, nil)
+	require.NoError(t, err, "AddVerifier failed")
+	res, err := f.FullNode.StateWaitMsg(ctx, sm.Cid(), 1, lapi.LookbackNoLimit, true)
+	require.NoError(t, err)
+	require.EqualValues(t, 0, res.Receipt.ExitCode)
```

Comment:
I think the datacap setup should go under `framework`, as it is not really important for the DDO test, and we'll probably use it elsewhere in the future too.

Reply:
Makes sense, I will refactor